great_expectations/data_context/types/base.py

```diff
@@ -1049,6 +1055,11 @@ def validate_schema(self, data, **kwargs):
     # noinspection PyUnusedLocal
     @post_load
     def make_datasource_config(self, data, **kwargs):
+        # Add names to data connectors
+        for data_connector_name, data_connector_config in data.get(
+            "data_connectors", {}
+        ).items():
+            data_connector_config["name"] = data_connector_name
```

Comment:
If the answer is yes that's fine but...
Do we want to be doing this in-place mutation of `data`?